rust/actyx/ax-core/src/runtime/operation/validation.rs

```diff
@@ -379,15 +420,17 @@ mod test {
             builder.encode_u64(10).encode_u64(100);
         });
         let cbor = cbor.decode();
-        assert!(validate_tuple(
-            &cbor,
-            &NonEmptyVec::try_from(vec![
-                Type::Atom(TypeAtom::Number(None)),
-                Type::Atom(TypeAtom::Number(None))
-            ])
-            .expect("a non empty vec")
-        )
-        .is_ok());
+        assert!(
+            validate_tuple(
+                &cbor,
+                &[
+                    Type::Atom(TypeAtom::Number(None)),
+                    Type::Atom(TypeAtom::Number(Some(100)))
+                ]
+            )
+            .is_ok(),
+            "TUPLE contents are not (NUMBER, NUMBER(100)"
```

Comment:
We probably talked past each other: this message doesn’t help when the test fails — what I need to know in that case is the error message (and possibly what the `&cbor` actually contained). I still think that `.unwrap()` is the best way to get this information.

Reply:
Reading this I’m quite sure we’re talking past each other. What I mean is this:
```rust
fn test_validate_tuple() {
    let cbor = CborBuilder::new().encode_array(|builder| {
        builder.encode_u64(10).encode_u64(100);
    });
    let cbor = cbor.decode();
    validate_tuple(&cbor, &[Atom(Number(None)), Atom(Number(Some(100)))]).unwrap();
}
```